Framework/DataHandling/src/LoadMuonNexusV2Helper.cpp

```diff
@@ -94,22 +104,27 @@ std::string loadMainFieldDirectionFromNexus(const NeXus::NXEntry &entry) {
   }
   return mainFieldDirection;
 }
+// Loads dead times from the nexus file
+// Assumes one grouping entry per detector
 std::vector<double>
 loadDeadTimesFromNexus(const NeXus::NXEntry &entry,
                        const std::vector<detid_t> &loadedDetectors,
-                       const bool isFileMultiPeriod) {
-
+                       const bool isFileMultiPeriod, const int periodNumber) {
+  // We cast the numLoadedDetectors to an int, which is the index type used for
+  // Nexus Data sets. As loadedDectors is derived from a nexus entry we
+  // can be certain we won't overflow the integer type.
+  int numLoadedDetectors = static_cast<int>(loadedDetectors.size());
   std::vector<double> deadTimes;
-  // Open detector nexus entry
+  deadTimes.reserve(numLoadedDetectors);
   NXClass detectorGroup = entry.openNXGroup(NeXusEntry::DETECTOR);
   if (detectorGroup.containsDataSet(NeXusEntry::DEADTIME)) {
     NXFloat deadTimesData = detectorGroup.openNXFloat(NeXusEntry::DEADTIME);
     deadTimesData.load();
-    if (!isFileMultiPeriod) {
-      // Simplest case - one grouping entry per detector
-      for (const auto &detectorNumber : loadedDetectors) {
-        deadTimes.emplace_back(deadTimesData[detectorNumber - 1]);
-      }
+    int deadTimeOffset =
+        !isFileMultiPeriod ? 0 : (numLoadedDetectors) * (periodNumber - 1);
+    for (const auto &detectorNumber : loadedDetectors) {
+      deadTimes.emplace_back(
+          deadTimesData[detectorNumber - 1 + deadTimeOffset]);
```

Comment:
This code is common to both so should move to a private method in a class as per above

Reply:
So the code is slightly different as deadTimesData is a nexus float entry whereas grouping is a nexus int entry. Grouping information hasn't been written in these new nexus files as of yet, so there’s a good chance the implementation will change (for instance if a detector maps to more than one group) so keeping them separate may be best for now.